liquibase-standard/src/main/java/liquibase/changelog/ChangeLogHistoryServiceFactory.java

```diff
@@ -3,80 +3,70 @@
 import liquibase.Scope;
 import liquibase.database.Database;
 import liquibase.exception.UnexpectedLiquibaseException;
-
-import java.util.*;
+import liquibase.plugin.AbstractPluginFactory;
+import liquibase.plugin.Plugin;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
-public class ChangeLogHistoryServiceFactory {
-
-    private static ChangeLogHistoryServiceFactory instance;
-
-    private final List<ChangeLogHistoryService> registry = new ArrayList<>();
+public class ChangeLogHistoryServiceFactory extends AbstractPluginFactory<ChangeLogHistoryService> {
 
+    private final List<ChangeLogHistoryService> explicitRegistered = new ArrayList<>();
     private final Map<Database, ChangeLogHistoryService> services = new ConcurrentHashMap<>();
 
+    @Deprecated // use Scope instead
```

Comment:
Could we make this deprecation have the standard JavaDocs formatting and information, please?

CC @filipelautert @rberezen